src/utils/createWeeklyList.ts

```diff
@@ -0,0 +1,162 @@
+  reportData.forEach((report) => {
+    combinedData.push(addTypeAndYearMonthDate(report, 'Report'));
+  });
+
+  mediaData.forEach((media) => {
+    combinedData.push(addTypeAndYearMonthDate(media, 'Media'));
+  });
+
+  combinedData.sort(sortData);
+
+  return combinedData;
+};
+
+const multiDimensionalUnique = (dateList: string[][]) => {
+  const uniques = [];
+  const itemsFound: { [key: string]: boolean } = {};
+
+  for (const date of dateList) {
+    const stringified = JSON.stringify(date);
+    if (itemsFound[stringified]) continue;
+    uniques.push(date);
+    itemsFound[stringified] = true;
+  }
+
+  return uniques;
+};
+
+const getStartDateOfWeek = (date: string) => {
+  let startDate = new Date(date);
+  if (isMonday(startDate))
+    return [
+      format(startDate, 'yyyy-MM-dd'),
+      format(startDate, 'yyyy년MM월dd일'),
+    ];
+
+  return [
+    format(previousMonday(startDate), 'yyyy-MM-dd'),
+    format(previousMonday(startDate), 'yyyy년MM월dd일'),
+  ];
+};
+
+const getEndDateOfWeek = (date: string) => {
+  let endDate = new Date(date);
+  if (isSunday(endDate))
+    return [format(endDate, 'yyyy-MM-dd'), format(endDate, 'yyyy년MM월dd일')];
+
+  return [
+    format(nextSunday(endDate), 'yyyy-MM-dd'),
+    format(nextSunday(endDate), 'yyyy년MM월dd일'),
+  ];
+};
+
+const makeWeekList = (date: string) => {
+  const [startDate, startDateOfWeek] = getStartDateOfWeek(date);
+  const [endDate, endDateOfWeek] = getEndDateOfWeek(date);
+  const periodOfWeek = `${startDateOfWeek} ~ ${endDateOfWeek}`;
+
+  return { periodOfWeek, startDate, endDate };
+};
+
+const makeDeduplicatedDateList = (combinedData: CombineDataType[]) => {
+  const dateList: string[][] = [];
+  [...new Set(combinedData.map((data) => data.date))].forEach((data) => {
+    const { periodOfWeek, startDate, endDate } = makeWeekList(data);
+    dateList.push([periodOfWeek, startDate, endDate]);
+  });
+
+  return multiDimensionalUnique(dateList);
+};
+
+const checkBetweenStartAndEndDate = ({
+  date,
+  startDate,
+  endDate,
+}: {
+  date: string;
+  startDate: string;
+  endDate: string;
+}) => {
+  return (
+    getUnixTime(new Date(date)) >= getUnixTime(new Date(startDate)) &&
+    getUnixTime(new Date(date)) <= getUnixTime(new Date(endDate))
+  );
+};
+
+const setWeeklyData = (
+  dataList: string[][],
+  combinedData: CombineDataType[]
+) => {
+  const weeklyData: TotalDataInterface = {};
+
+  dataList.forEach(([periodOfWeek, startDate, endDate]) => {
+    const reportData = combinedData.filter(
+      (data) =>
+        data.dataType === 'Report' &&
+        checkBetweenStartAndEndDate({ date: data.date, startDate, endDate })
+    );
+
+    const mediaData = combinedData.filter(
+      (data) =>
+        data.dataType === 'Media' &&
+        checkBetweenStartAndEndDate({ date: data.date, startDate, endDate })
+    );
+
+    weeklyData[periodOfWeek] = {
+      reports: reportData as ReportInterface[],
+      media: mediaData as MediaInterface[],
+    };
```

Comment:
요렇게 수정한다면 filter로 가져오는 단계에서 타입체크를 하여 이상한 데이터가 변수에 들어가는 것을 방지해줄 것 같습니다
```suggestion
    const reportData: ReportInterface[] = combinedData.filter(
      (data) =>
        data.dataType === 'Report' &&
        checkBetweenStartAndEndDate({ date: data.date, startDate, endDate })
    );

    const mediaData: MediaInterface[] = combinedData.filter(
      (data) =>
        data.dataType === 'Media' &&
        checkBetweenStartAndEndDate({ date: data.date, startDate, endDate })
    );

    weeklyData[periodOfWeek] = {
      reports: reportData,
      media: mediaData,
    };
```

Reply:
이건 봤는데 생각해보니 combineData가 (ReportInterface | MediaInterface)[] 타입이라 타입단언을 해야 될 거 같습니다.